packages/flutter/lib/src/widgets/actions.dart

```diff
@@ -507,12 +530,20 @@ class Actions extends StatefulWidget {
   /// Creates a dependency on the [Actions] widget that maps the bound action so
   /// that if the actions change, the context will be rebuilt and find the
   /// updated action.
-  static Action<T>? find<T extends Intent>(BuildContext context, {bool nullOk = false}) {
+  ///
+  /// The optional `intent` argument supplies the type of the intent to look for
+  /// if the concrete type of the intent sought isn't available. If not
+  /// supplied, then `T` is used.
+  static Action<T>? find<T extends Intent>(BuildContext context, {bool nullOk = false, T? intent}) {
     Action<T>? action;
+    // Specialize the type if a runtime example instance of the intent is given.
+    // This allows this function to be called by code that doesn't know the
+    // concrete type of the intent at compile time.
+    final Type type = intent?.runtimeType ?? T;
```

Comment:
Wondering whether we should assert here that type is not  equal to `Intent`. If it is Intent, I think somebody didn't specify either T or `intent` and that's probably an error given that you can't have an Intent intent?

Reply:
On the other hand, it seems you can instantiate `Intent`, it's not abstract. But maybe it should be abstract?

Reply:
No, we definitely shouldn't assert.  The reason is that it will almost certainly not be equal to `Intent` if we don't know the type a compile time.

Consider the use case where I use that `intent` argument:
```dart
      final Action<Intent>? action = Actions.find<Intent>(
        primaryContext,
        intent: matchedIntent,
        nullOk: true,
      );
```

`matchedIntent` almost certainly does not have a runtime type of `Intent`, and that's why we have to supply it, so that we can get at the runtime type.

Also, if we could assert that the type was equal to `Intent`, we wouldn't need to have this argument at all, since the author would know the type to pass to the generic: `Intent`.

I'll look at making `Intent` abstract though, that might make sense.

Reply:
Aaaaand I missed the "not" in "not equal to Intent" of your comment. :-)  OK, looking at that as a possibility.

Reply:
Wait, I meant, can we assert that it is **not equal** to Intent. Your explanation seems to support that assert because:

>  The reason is that it will almost certainly not be equal to Intent

What I am saying is, that it 100% should not be equal to Intent.

Reply:
Yeah. Reading carefully: I should try it sometime. :-)

OK, I added the assert, good suggestion.